api/src/main/scala/pl/touk/nussknacker/engine/api/process/RunMode.scala

```diff
@@ -3,16 +3,21 @@ package pl.touk.nussknacker.engine.api.process
 /**
   * Specifies the mode a node is invoked. It can be one of the following values:
   * <ul>
-  * <li>Normal - process is executed inside Flink or other engine. Real implementation of node should be used. Note, this
-  * mode is also used for service query, validations and test data generation.</li>
+  * <li>Normal - used when process is executed inside Flink or other engine, and also for service query, validations and test data generation.
+  * In this mode real implementation of node should be used.</li>
   * <li>Test - process is run in test mode to collect results for test data. Can be used to stub real implementation.</li>
   * </ul>
   */
 sealed trait RunMode
 
 object RunMode {
 
-  case object Normal extends RunMode
+  sealed trait Normal extends RunMode
+
+  case object Engine extends Normal
+  case object Validation extends Normal
```

Comment:
Validation doesn't cause that node will be "run". What case do you see that user will need to handle it in component?

Reply:
Actually I see that from diffrent angle - I care only about `Engine` mode, to be able to distinguish that my code is run via engine vs designer. 
What mode do you think should I use while performing validation?

Reply:
We agree that the real usage is not only for run part but also for definition part and runMode is passed that way to both places. So the only problem was Run in name of class. Let's rename it to ComponentUsage or similar